### PR TITLE
[WIP] Added support for remote cluster

### PR DESCRIPTION
**Pull request by evalsocket**
- Added support for remote cluster

### CODE REVIEW

**Comment by evalsocket** [comment]
We need `flyte_workflows_register` independent to `docker-compose`

File: src/pytest_flyte/plugin.py [hunk 1]
```diff
@@ -124,19 +124,25 @@ def docker_cleanup():
...
-    with capsys_suspender():
-        docker_services.wait_until_responsive(timeout=900, pause=1, check=_check)
-    return friendly.SynchronousFlyteClient(url, insecure=True)
+
+    if "FLYTE_PLATFORM_INSECURE" not in os.environ:
+        os.environ["FLYTE_PLATFORM_INSECURE"] = "true"
+
+    if "FLYTE_PLATFORM_URL" in os.environ:
+        return friendly.SynchronousFlyteClient(os.environ["FLYTE_PLATFORM_URL"], insecure=bool(os.environ["FLYTE_PLATFORM_INSECURE"]))
+    else:
```
Comment (cosmicBboy) [inline src/pytest_flyte/plugin.py:133]: nit: else block not needed due to early return in line 132 above

File: src/pytest_flyte/plugin.py [hunk 2]
```diff
@@ -124,19 +124,25 @@ def docker_cleanup():
...
-
-    with capsys_suspender():
-        docker_services.wait_until_responsive(timeout=900, pause=1, check=_check)
-    return friendly.SynchronousFlyteClient(url, insecure=True)
+
+    if "FLYTE_PLATFORM_INSECURE" not in os.environ:
+        os.environ["FLYTE_PLATFORM_INSECURE"] = "true"
+
+    if "FLYTE_PLATFORM_URL" in os.environ:
+        return friendly.SynchronousFlyteClient(os.environ["FLYTE_PLATFORM_URL"], insecure=bool(os.environ["FLYTE_PLATFORM_INSECURE"]))
```
Comment (cosmicBboy) [inline src/pytest_flyte/plugin.py:132]: why not `insecure=True` directly?

File: src/pytest_flyte/plugin.py [hunk 3]
```diff
@@ -124,19 +124,25 @@ def docker_cleanup():
...
+            try:
+                docker_compose.execute("exec backend wait-for-flyte.sh")
+                return True
+            except Exception as e:
+                print(e)
+                return False
+
+        with capsys_suspender():
+            docker_services.wait_until_responsive(timeout=900, pause=1, check=_check)
+        return friendly.SynchronousFlyteClient(url, insecure=bool(os.environ["FLYTE_PLATFORM_INSECURE"]))
```
Comment (cosmicBboy) [inline src/pytest_flyte/plugin.py:148]: why not `insecure=True` directly?

File: src/pytest_flyte/plugin.py (hunk 2 above)
Comment (cosmicBboy) [inline src/pytest_flyte/plugin.py:132]: consider using `pytest_addoption`, e.g. https://docs.pytest.org/en/stable/writing_plugins.html#testing-plugins, something like:

```python
def pytest_addoption(parser):
    group = parser.getgroup("helloworld")
    group.addoption(
        "--flyte-platform-url",
        action="store",
        dest="flyte_platform_url",
        default=None,
        help='Default "name" for hello().',
    )


def flyteclient(request, ...):
    flyte_platform_url = request.config.getoption("flyte_platform_url")
    if flyte_platform_url is not None:
        return friendly.SynchronousFlyteClient(flyte_platform_url, insecure=True)
```

**Pull request closed without merging**